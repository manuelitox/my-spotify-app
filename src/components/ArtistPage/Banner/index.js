import React from 'react'

import StyledBanner from './styles'
import BannerWrapper from './wrapper'
import StylesWaveArtistIcon from './stylesWaiveIcon'

const Banner = () => (
  <StyledBanner>
    <BannerWrapper />
    <StylesWaveArtistIcon />
  </StyledBanner>
)

export default Banner