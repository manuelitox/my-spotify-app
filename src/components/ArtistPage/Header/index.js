import React from 'react'
import { Link } from 'react-router-dom'

import StyledHeader from './styles'
import SearchFormContainer from 'components/generic/SearchForm/container'
import { ReactComponent as HeadsetIcon } from 'svgs/headset.svg'

const Header = () => (
  <StyledHeader>
    <Link to="/">
      <HeadsetIcon height="45" width="55" viewBox="0 0 70 55"/>
    </Link>
    <SearchFormContainer placeholder="search another artist..." />
  </StyledHeader>
)

export default Header