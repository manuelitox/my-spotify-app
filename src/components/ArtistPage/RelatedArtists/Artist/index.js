import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import StyledArtist from './styles'
import PlaceholderArtist from 'svgs/placeholderArtist.svg'

const Artist = ({ id, name, photo }) => (
  <StyledArtist>
    <img 
      alt={ name }
      src={ photo ? photo : PlaceholderArtist } 
      width="40"
      height="40" />
    <h4>{ name }</h4>
    <Link to={ `/artist/${ name.replace(' ', '-') }/${ id }` } />
  </StyledArtist>
)

Artist.propTypes =Â {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  photo: PropTypes.string,
}

export default Artist