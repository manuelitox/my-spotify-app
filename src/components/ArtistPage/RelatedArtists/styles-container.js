import styled from 'styled-components'

import WithContainer from 'components/generic/Container'

const StyledContainerRelatedArtists = styled.div`
  padding: 40px 0;
`

export default WithContainer(StyledContainerRelatedArtists)