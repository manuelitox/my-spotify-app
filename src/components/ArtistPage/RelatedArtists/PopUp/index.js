import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Artist from 'components/ArtistPage/RelatedArtists/Artist/'
import StyledWrapperRelatedArtists from 'components/ArtistPage/RelatedArtists/styles-wrapper'
import getImage from 'lib/get-image'

const PopUpRelatedArtists = ({ isOpen, togglePopUp, relatedArtists }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Related Artists</Title>
      <StyledWrapperRelatedArtists inPopup onClick={ togglePopUp }>
        { relatedArtists.map((artist, index) =>Â (
          <Artist key={ index } id={ artist.id } photo={ getImage(artist.images, 200) } name={ artist.name } />
        )) }   
      </StyledWrapperRelatedArtists>
    </Fragment>
  </PopUp>      
)

PopUpRelatedArtists.propTypes = {
  isOpen: PropTypes.bool,
  togglePopUp: PropTypes.func.isRequired,
  relatedArtists: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      images: PropTypes.array      
    })
  ).isRequired
}

export default PopUpRelatedArtists