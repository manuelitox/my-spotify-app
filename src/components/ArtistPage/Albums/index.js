import React, { Component } from 'react'

import Title from 'components/ArtistPage/Title/'
import Album from './Album/'
import PopUp from 'components/generic/PopUp/container'
import Button from 'components/generic/Button/'
import StyledWrapperAlbums from './styles'

class Albums extends Component {
  state = { isOpen: false }
  
  togglePopUp = () => this.setState({ isOpen: !this.state.isOpen })

  render () {
    return (
      <div>
        <Title>Albums</Title>
        <StyledWrapperAlbums>
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
          <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
        </StyledWrapperAlbums>
        <Button onClick={ this.togglePopUp }>view more albums</Button>
        <PopUp isOpen={ this.state.isOpen } togglePopUp={ this.togglePopUp }>
          <div>
            <h1>Albums list</h1>
            <p>description...</p>
          </div>
        </PopUp>
      </div>      
    )
  } 
}

export default Albums