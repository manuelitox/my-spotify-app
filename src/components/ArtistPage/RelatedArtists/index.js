import React, { Component } from 'react'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Button from 'components/generic/Button/'
import Artist from './Artist/'
import StyledRelatedArtists from './styles'
import StyledWrapperRelatedArtists from './styles-wrapper'
import StyledContainerRelatedArtists from './styles-container'

class RelatedArtists extends Component {
  state = { isOpen: false }
  
  togglePopUp = () => this.setState({ isOpen: !this.state.isOpen })

  render () {
    return (
      <StyledRelatedArtists>
        <StyledContainerRelatedArtists>
          <Title color={ theme.blueDark }>Related Artists</Title>
          <StyledWrapperRelatedArtists>
            <Artist id="1" name="Martin Abasto" />
            <Artist id="2" name="Bairam Frootan" />
            <Artist id="3" name="Emmalynn Mazzia" />
            <Artist id="4" name="Gabriel Pires" />
            <Artist id="5" name="Irene Sotelo" />
            <Artist id="6" name="Oea Romana" />
            <Artist id="7" name="Pin Jung-Eym" />
            <Artist id="8" name="Udom Paowsong" />    
            <Artist id="9" name="John Doe" />    
          </StyledWrapperRelatedArtists>
          <Button 
            bg={ theme.blueDark }
            textColor="white"
            onClick={ this.togglePopUp }>
            view related artists
          </Button>
          <PopUp isOpen={ this.state.isOpen } togglePopUp={ this.togglePopUp }>
            <div>
              <h1>Related Artists list</h1>
              <p>description...</p>
            </div>
          </PopUp>          
        </StyledContainerRelatedArtists>
      </StyledRelatedArtists>
    )
  }
}

export default RelatedArtists