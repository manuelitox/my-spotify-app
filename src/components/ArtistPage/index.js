import React from 'react'
 
import Header from './Header/'
import Banner from './Banner/'
import Albums from './Albums/'
import Tracks from './Tracks/'
import RelatedArtists from './RelatedArtists/'
import StyledArtistPage from './styles'
import StyledContainerAlbumsAndTopTracks from './stylesContainerAlbumsAndTopTracks'

const ArtistPage = () => (
  <StyledArtistPage>
    <Header />
    <Banner />
    <StyledContainerAlbumsAndTopTracks>
      <Albums />
      <Tracks />
    </StyledContainerAlbumsAndTopTracks>
    <RelatedArtists />
  </StyledArtistPage>
)

export default ArtistPage