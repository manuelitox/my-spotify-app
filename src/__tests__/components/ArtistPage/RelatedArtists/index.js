import React from 'react'
import { shallow } from 'enzyme'

import { RelatedArtists } from 'components/ArtistPage/RelatedArtists/'

describe('Component | RelatedArtists', () => {
  const relatedArtists = {
    artists: [
      {
        id: '123',
        name: 'artist name',
        images: [{ url: 'path/to/photo', height: 200 }]
      }
    ]
  }
  let component 
  beforeEach( () => {
    component = shallow(
      <RelatedArtists relatedArtists={ relatedArtists } />
    )
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
})