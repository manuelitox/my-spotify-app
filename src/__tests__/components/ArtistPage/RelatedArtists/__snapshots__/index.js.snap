// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | RelatedArtists should render a component 1`] = `
<styled.div>
  <Component>
    <Title
      color="#262D45"
    >
      Related Artists
    </Title>
    <styled.div>
      <Artist
        id="1"
        name="Martin Abasto"
      />
      <Artist
        id="2"
        name="Bairam Frootan"
      />
      <Artist
        id="3"
        name="Emmalynn Mazzia"
      />
      <Artist
        id="4"
        name="Gabriel Pires"
      />
      <Artist
        id="5"
        name="Irene Sotelo"
      />
      <Artist
        id="6"
        name="Oea Romana"
      />
      <Artist
        id="7"
        name="Pin Jung-Eym"
      />
      <Artist
        id="8"
        name="Udom Paowsong"
      />
      <Artist
        id="9"
        name="John Doe"
      />
    </styled.div>
    <Button
      bg="#262D45"
      onClick={[Function]}
      textColor="white"
    >
      view related artists
    </Button>
    <OnClickOutside(PopUpContainer)
      eventTypes={
        Array [
          "mousedown",
          "touchstart",
        ]
      }
      excludeScrollbar={false}
      isOpen={false}
      outsideClickIgnoreClass="ignore-react-onclickoutside"
      preventDefault={false}
      stopPropagation={false}
      togglePopUp={[Function]}
    >
      <div>
        <h1>
          Related Artists list
        </h1>
        <p>
          description...
        </p>
      </div>
    </OnClickOutside(PopUpContainer)>
  </Component>
</styled.div>
`;
